src/main/java/seedu/address/model/property/Address.java

```diff
@@ -9,13 +9,17 @@
  */
 public class Address {
 
-    public static final String MESSAGE_CONSTRAINTS = "Addresses can take any values, and it should not be blank";
+    public static final Integer MAX_LENGTH = 100;
+    public static final String MESSAGE_CONSTRAINTS = "Addresses should only contain alphanumeric characters, "
+            + "hashes (#), commas(,), semicolons (;), hyphens (-), and spaces, and it should not be blank.\n"
+            + "Addresses should also be at most " + MAX_LENGTH + " characters long.";
 
     /*
      * The first character of the address must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
+     * The other characters include hyphens, hashes and spaces.
      */
-    public static final String VALIDATION_REGEX = "[^\\s].*";
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}\\-][\\p{Alnum}\\-#,; ]{0," + (MAX_LENGTH - 1) + "}";
```

Comment:
Is it intentional that the address allows hyphen as the first character?

Reply:
Yes, so that the address "-" is valid